cpp/CMakeLists.txt

```diff
@@ -434,6 +437,10 @@ if(RAFT_COMPILE_LIBRARY)
   # ensure CUDA symbols aren't relocated to the middle of the debug build binaries
   target_link_options(raft_lib PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/fatbin.ld")
 
+  # Optimization: extra compile flags for individual compilation units
+  file(GLOB REFINE_HOST_SRC_FILES "src/neighbors/detail/refine_host_*.cpp")
+  set_property(SOURCE ${REFINE_HOST_SRC_FILES} PROPERTY COMPILE_FLAGS "-ftree-vectorize")
+
```

Comment:
Note, I added these lines to cmake to do extra optimization and enable vectorization during my experiments. At some point, we could add here some flags to enable x86 avx or arm sve.  I'd like to hear any thoughts of @rapidsai/raft-cmake-codeowners, whether this is considered a good practice. Otherwise, I'd remove this as not critical.

Reply:
Specfying `-ftree-vectorize` isn't needed as it is part of the set of flags that `-O3` sets which is the optimization level we set when compiling in Release mode.

As far as setting SVE I am extremely hesistant as lots of arm nodes don't support that ( e.g. only AWS Graviton3 supports it from AWS ).

Reply:
AVX2 is availabe on virtually all processors used by the raft users. SVE is supported by Grace, so it may be worth adding a separate knob for that.
In the RelWithDebugInfo mode only O2 is used, which does not include 'tree-vectorize'. This is very handy during development and experimentation. Besides, there's no harm in specifying a flag multiple times, I've put it there more for humans to read and being aware that he vectorizarion is important for that bit of code.

Apart from the problems with specific flags, is a construct like this a bad style for a cmake project?

Reply:
> Apart from the problems with specific flags, is a construct like this a bad style for a cmake project?
Yes, all sources should be listed explicitly.

> AVX2 is availabe on virtually all processors used by the raft users. SVE is supported by Grace, so it may be worth adding a separate knob for that

These very specific platform flags should be set in a toolchain file that best describes the platform. This toolchains can be selected by the build-infra when it needed. That way we don't publish packages that don't execute on a subset of supported hardware due to a misconfigured default.

We shouldn't conditionally add optimizations like these to a subset of files without first showing why these optimizations fail for the entire project. If somebody wants AVX2 or SVE I see no reason why they wouldn't want that for all of RAFT.

Reply:
Makes sense, thanks for the clarification. My particular use was was to isolate the optimization during the debugging to isolate the bug I faced. I guess this is too specific of a use case to keep it in CMakeLists indefinitely.